### PR TITLE
parser: add collations from MySQL 8.0

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/7968

Problem Summary: 

DM failed to parse CREATE TABLE when upstream is MySQL 8.0, like https://asktug.com/t/topic/573639

### What is changed and how it works?

add parser support for collations in MySQL 8.0.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [x] Changes MySQL compatibility

### Release note

```release-note
Enhance compatibility of parsing collations of MySQL 8.0
```

### CODE REVIEW

**Review by xiongjiwei — APPROVED** [review]

File: parser/parser_test.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 	"testing"
 
 	"github.com/pingcap/errors"
+	"github.com/stretchr/testify/require"
+
```
Comment (wjhuang2016) [inline parser/parser_test.go:25]: 。。。

Comment (lance6716) [reply parser/parser_test.go:25]: what's the rule of separating 3-rd party imports? parser/charset/charset.go did so but this file didn't

Comment (wjhuang2016) [reply parser/parser_test.go:25]: We don't care about it.

**Review by lcwangchao — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**